python/packages/ai/teams/ai/state/message.py

```diff
@@ -0,0 +1,15 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+from dataclasses import dataclass
+from typing import Literal
+
+MessageRole = Literal["system", "user", "assistant", "say", "do"]
```

Comment:
this was a suggestion made by Steve, instead of using the plain text conversation history implementation used in the other SDKs (which we will likely change soon) we are using a message object that we based off the one OpenAI uses https://platform.openai.com/docs/guides/gpt so we can more easily query history. DO and SAY are messages that may occur in the conversation history and we want the developers to have an interface for accessing them.

Reply:
Agree that using the data models from OpenAI is better. But the document mentions the available roles are `system`, `user` and `assistant`. So seems `say` and `do` should not in the list?

Reply:
Ya thats our own custom stuff because the `ConversationHistory` class needs to be able to do things like "get the last `say` message from history and replace it". I created this `Message` class to help do these queries in a more reliable way than using an array of strings like the JS implementation does because after talking with Steve he said we intend to make a similar change in the other libraries anyway. 

Do you have any ideas how we could do this better than what I've added in this PR? I am open to alternative approaches and maybe we can discuss this more in our weekly sync. If we come up with something better we can create a work item to improve upon or redo this history class entirely.

Reply:
Thank you for the explanation. I fine with it now by knowing the reason of having "say" and "do".